letsencrypt-auto-source/tests/oraclelinux6_tests.sh

```diff
@@ -27,55 +24,31 @@ if [ $RESULT -ne 0 ]; then
   exit 1
 fi
 
-echo "PASSED: bootstrap from old letsencrypt-auto succeeded and installed Python 3.4"
+echo "PASSED: Bootstrap from old letsencrypt-auto succeeded and installed Python 3.4"
 
-# Expect letsencrypt-auto to just fail to rebootstrap and start certbot in interactive
-# shell since SCL is not installed.
-if ! "$LE_AUTO" --version 2>&1 | grep -q "Enable the SCL repository and try running Certbot again."; then
+# Expect letsencrypt-auto to skip rebootstrapping with a warning since SCL is not installed.
+if ! "$LE_AUTO" --non-interactive --version 2>&1 | grep -q "Skipping upgrade because new OS dependencies may need to be installed."; then
   echo "FAILED: Script letsencrypt-auto managed to start Certbot in interactive shell while SCL is not enabled!"
   exit 1
 fi
 
-echo "PASSED: Script letsencrypt-auto did not rebootstrap and did not star Certbot."
-
-# Expect letsencrypt-auto to not update certbot, but start it anyway in non-interactive
-# shell since SCL is not installed.
-# NB: Readline has an issue on all Python versions for CentOS 6, making `certbot --version`
-# output an unprintable ASCII character on a new line at the end.
-# So we take the second last line of the output.
-version=$($LE_AUTO --version 2>/dev/null | tail -2 | head -1)
-
-if [ -z "$version" ]; then
-  echo "ERROR: Script letsencrypt-auto failed to start certbot in a non-interactive shell while SCL was not enabled."
-  exit 1
-fi
-
-if [ "$version" != "$INITIAL_CERTBOT_VERSION_PY34" ]; then
-  echo "ERROR: Script letsencrypt-auto upgraded certbot in a non-interactive shell while SCL was not enabled."
-  exit 1
-fi
-
-echo "PASSED: Script letsencrypt-auto did not upgraded certbot but started it successfully while SCL was not enabled."
+echo "PASSED: Script letsencrypt-auto did not rebootstrap."
 
 # Enable SCL
 yum install -y oracle-softwarecollection-release-el6 >/dev/null
 
-# Following test is exectued in a subshell, to not leak any environment variable
-(
-  export VENV_PATH=$(mktemp -d)
 
-  # Expect letsencrypt-auto to bootstrap successfully since SCL is available
-  "$LE_AUTO" -n >/dev/null 2>/dev/null
+# Expect letsencrypt-auto to bootstrap successfully since SCL is available.
+"$LE_AUTO" -n --version >/dev/null 2>/dev/null
 
-  if [ "$($VENV_PATH/bin/python -V 2>&1 | cut -d" " -f2 | cut -d. -f1-2)" != "3.6" ]; then
-    echo "ERROR: Script letsencrypt-auto failed to bootstrap and install Python 3.6 while SCL is available."
-    exit 1
-  fi
+if [ "$(/opt/eff.org/certbot/venv/bin/python -V 2>&1 | cut -d" " -f2 | cut -d. -f1-2)" != "3.6" ]; then
+  echo "ERROR: Script letsencrypt-auto failed to bootstrap and install Python 3.6 while SCL is available."
+  exit 1
+fi
 
-  if ! $VENV_PATH/bin/certbot --version > /dev/null 2> /dev/null; then
-    echo "ERROR: Script letsencrypt-auto did not install certbot correctly while SCL is enabled."
-    exit 1
-  fi
-)
+if ! /opt/eff.org/certbot/venv/bin/certbot --version > /dev/null 2> /dev/null; then
```

Comment:
Can we extend this test just a bit since we won't be testing on RHEL/OL/SL 6 in our test farm by doing something similar to:
https://github.com/certbot/certbot/blob/db46326e9597f2166483f5da459069fe8afd509c/tests/letstest/scripts/test_leauto_upgrades.sh#L108-L109

Reply:
This will not work, most of the time. Indeed `certbot-auto` does not update the Certbot instance when its version (`certbot-auto` one) is a development version (like `0.40.dev0` currently).

Reply:
My linked code runs nightly, but the reason we can't take the same approach here which I didn't see before is that we don't have access to `certbot-auto`/`letsencrypt-auto` in the parent directory. With this in mind, let's ignore my suggestion :smile: